### PR TITLE
Tracer Advection on a Conformal Cubed Sphere Grid

### LINKED ISSUE #3204: Advection of tracer with prescribed or constant velocities over a `ConformalCubedSphereGrid`
Add a validation script and/or tests.

### PR DESCRIPTION
**Pull request by siddharthabishnu**
This PR closes #3204.

### CODE REVIEW

File: validation/multi_region/multi_region_tracer_advection.jl [hunk 1]
```diff
@@ -87,31 +154,28 @@ simulation.callbacks[:save_tracer] = Callback(save_tracer, TimeInterval(Δt))
...
 
-GLMakie.record(fig, "multi_region_tracer_advection.mp4", frames, framerate = 1) do i
-    msg = string("Plotting frame ", i, " of ", frames[end])
-    print(msg * " \r")
-    if plot_type == :heatsphere
-        heatsphere!(ax, tracer_fields[i]; colorrange = (-1, 1), colormap = :balance)
-    elseif plot_type == :heatlatlon
-        heatlatlon!(ax, tracer_fields[i]; colorrange = (-1, 1), colormap = :balance)
-    end
-end
```
Comment (navidcy) [inline validation/multi_region/multi_region_tracer_advection.jl]: @siddharthabishnu this was making another plot for every frame. What I did just changes the array plotted and it's much much much faster.

Comment (siddharthabishnu) [reply validation/multi_region/multi_region_tracer_advection.jl]: I see. The plotting is indeed much faster now. Thanks.

File: src/MultiRegion/multi_region_abstract_operations.jl [hunk 2]
```diff
@@ -24,7 +24,7 @@ sync_all_devices!(f::MultiRegionAbstractOperation)  = sync_all_devices!(devices(
 @inline switch_device!(f::MultiRegionAbstractOperation, d) = switch_device!(f.grid, d)
 @inline getdevice(f::MultiRegionAbstractOperation, d)      = getdevice(f.grid, d)
 
-for T in [:BinaryOperation, :UnaryOperation, :MultiaryOperation, :Derivative, :ConditionalOperation]
+for T in [:BinaryOperation, :UnaryOperation, :MultiaryOperation, :Derivative]
```
Comment (navidcy) [inline src/MultiRegion/multi_region_abstract_operations.jl:27]: @siddharthabishnu, why this change?

File: src/MultiRegion/multi_region_cubed_sphere_grid.jl [hunk 3]
```diff
@@ -210,41 +210,100 @@ function ConformalCubedSphereGrid(arch::AbstractArchitecture=CPU(), FT=Float64;
...
                                                                                            devices)
 
-    fields₁ = (:Δxᶜᶜᵃ,  :Δxᶠᶜᵃ,  :Δxᶜᶠᵃ,  :λᶜᶜᵃ,   :λᶠᶜᵃ,   :λᶜᶠᵃ,   :Azᶜᶜᵃ,  :Azᶠᶜᵃ)
-    LXs₁    = (:Center, :Face,   :Center, :Center, :Face,   :Center, :Center, :Face)
-    LYs₁    = (:Center, :Center, :Face,   :Center, :Center, :Face,   :Center, :Center)
+    # the following fills the coordinate and metric halos using halo-filling algorithms
+    ccacoords = (:λᶜᶜᵃ, :φᶜᶜᵃ)
+    fcacoords = (:λᶠᶜᵃ, :φᶠᶜᵃ)
+    cfacoords = (:λᶜᶠᵃ, :φᶜᶠᵃ)
+    ffacoords = (:λᶠᶠᵃ, :φᶠᶠᵃ)
```
Comment (navidcy) [inline src/MultiRegion/multi_region_cubed_sphere_grid.jl:217]: @siddharthabishnu you seem to undo here the fixes I made in #3256

Comment (siddharthabishnu) [reply src/MultiRegion/multi_region_cubed_sphere_grid.jl:217]: After you merged the PR associated with #3256, I merged the main branch into this one, and accepted the 'incoming changes'. I just wanted to incorporate those changes, not undo them. I am not sure how that happened.

Comment (navidcy) [reply src/MultiRegion/multi_region_cubed_sphere_grid.jl:217]: Okie then! No worries

**Comment by siddharthabishnu** [comment]
Closing this PR since this work is being continued on #3266.
